examples/vae/main.py

```diff
@@ -0,0 +1,147 @@
+from absl import app
+from absl import flags
+
+import jax.numpy as jnp
+import numpy as np
+
+import jax
+from jax import nn as jnn
+from jax import random
+
+from flax import nn
+from flax import optim
+
+import tensorflow_datasets as tfds
+
+from utils import save_image
+
+
+EPS = 1e-8
+KEY = random.PRNGKey(0)
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_float(
+    'learning_rate', default=1e-3,
+    help=('The leanring rate for the Adam optimizer')
+)
+
+flags.DEFINE_integer(
+    'batch_size', default=128,
+    help=('Batch size for training')
+)
+
+flags.DEFINE_integer(
+    'num_epochs', default=10,
+    help=('Number of training epochs')
+)
+
+
+class Encoder(nn.Module):
+    def apply(self, x):
+        x = nn.Dense(x, 400, name='enc_fc1')
+        x = jnn.relu(x)
+        mean_x = nn.Dense(x, 20, name='enc_fc21')
+        logvar_x = nn.Dense(x, 20, name='enc_fc22')
+        return mean_x, logvar_x
+
+
+class Decoder(nn.Module):
+    def apply(self, z):
+        z = nn.Dense(z, 400, name='dec_fc1')
+        z = jnn.relu(z)
+        z = nn.Dense(z, 784, name='dec_fc2')
+        z = jnn.sigmoid(z)
```

Comment:
I think log_sigmoid is more stable?

Reply:
I decided to remove this line to make `decoder` return `logits`.